runtime-modules/content/src/permissions/curator_group.rs

```diff
@@ -1,4 +1,48 @@
 use super::*;
+use sp_std::collections::btree_map::BTreeMap;
+#[cfg(feature = "std")]
+use strum_macros::EnumIter;
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+pub struct ContentModerationPermissions {
+    pub hide_video: bool,
+    pub hide_channel: bool,
+    pub pause_channel: bool,
+    pub delete_object: bool,
+    pub delete_nft_free_video: bool,
+    pub delete_any_video: bool,
+    pub delete_channel: bool,
+}
+
+impl Default for ContentModerationPermissions {
+    fn default() -> Self {
+        Self {
+            hide_video: false,
+            hide_channel: false,
+            pause_channel: false,
+            delete_object: false,
+            delete_nft_free_video: false,
+            delete_any_video: false,
+            delete_channel: false,
+        }
+    }
+}
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize, EnumIter))]
+#[derive(Encode, Decode, Clone, Copy, PartialEq, Eq, Debug)]
+pub enum ContentModerationAction {
+    HideVideo,
+    HideChannel,
+    PauseChannel,
```

Comment:
Here it was meant that pause controls were more granular, as the requirements alluded

> Critically, an invocation of this pause command does not need to block all of these, it can be issued to only block a subset, as needed.

So you may as well make life simple and just flatten all cases directly into `ContentModerationAction`.

Reply:
Introduced `PausedChannelFunctionality` enum in https://github.com/Joystream/joystream/pull/3366/commits/001745b49cb9fd57fd3c41646c95e172635ae6e5